component/src/test/java/org/wso2/extension/siddhi/io/http/sink/HttpOAuthFailTestCase.java

```diff
@@ -0,0 +1,445 @@
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.extension.siddhi.io.http.sink;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.extension.siddhi.io.http.sink.util.HttpOAuthListenerHandler;
+import org.wso2.extension.siddhi.map.xml.sinkmapper.XMLSinkMapper;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class HttpOAuthFailTestCase {
+    private static final Logger log = Logger.getLogger(HttpAuthTestCase.class);
+    public static final String CARBON_HOME = "carbon.home";
+    private AtomicInteger eventCount = new AtomicInteger(0);
+
+    /**
+     *Set carbon-home
+     */
+    private void setCarbonHome() {
+        Path carbonHome = Paths.get("");
+        carbonHome = Paths.get(carbonHome.toString(), "src", "test");
+        System.setProperty(CARBON_HOME, carbonHome.toString());
+        log.info("Carbon Home Absolute path set to: " + carbonHome.toAbsolutePath());
+    }
+
+    /**
+     * Creating test for publishing events with valid access token.
+     *
+     * @throws Exception Interrupted exception
+     */
+    @Test
+    public void testHTTPTextOAuthTrue() throws Exception {
+        setCarbonHome();
+        log.info("Creating test for publishing events with valid access token.");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.setExtension("xml-output-mapper", XMLSinkMapper.class);
+        String inStreamDefinition = "Define stream FooStream (message String,method String,headers String);" +
+                "@sink(type='http',publisher.url='https://localhost:8015/abc',method='{{method}}'" +
+                ",headers='{{headers}}',consumer.key='addConsumerKey', consumer.secret='addConsumerSecret', " +
+                "token.url='https://localhost:8005/token', @map(type='xml', @payload('{{message}}'))) "
+                + "Define stream BarStream (message String,method String,headers String);";
+        String query = (
+                "@info(name = 'query') "
+                        + "from FooStream "
+                        + "select message,method,headers "
+                        + "insert into BarStream;"
+        );
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+        InputHandler fooStream = siddhiAppRuntime.getInputHandler("FooStream");
+        siddhiAppRuntime.start();
+        HttpOAuthListenerHandler lst = new HttpOAuthListenerHandler(8005, 8015);
```

Comment:
Let's put a self-explanatory name other than `lst`. e.g. listenerHandler or something like that.